[KIALI-153] Show Graph legend

aljesusg opened the pull request:
![legend](https://user-images.githubusercontent.com/3019213/40498227-baaea556-5f7e-11e8-8fd4-3608c977223e.gif)

Still working, please feedback ^^ @abonas @serenamarie125 @xeviknal

aljesusg commented:
@serenamarie125 related with colours, Is there any place where patternfly colors are defined?

abonas commented:
I think that the "show legend" should not be hidden in the "..." part, most of the users won't understand it is there. There is enough room for "hide/show legend" imo in the toolbar.

abonas commented:
and yes, the colors should match PF and should match the graph itself :)

aljesusg commented:
@abonas I discussed this with @xeviknal , the proposal of serena was this, I don't know why the button is inside the dropdownkebap, I'll change
![legend](https://user-images.githubusercontent.com/3019213/40484598-70d12c28-5f5c-11e8-9fdc-2fc8d1ce0f4c.png)

About colors ....yes ^^ I need to change them. Thanks

abonas commented:
@serenamarie125 your feedback is appreciated :)

aljesusg commented:
Updated Legend, changed to button and PF Colors:
```javascript
const PFCOLORS = {
  'pf-red-100': '#cc0000',
  'pf-orange-300': '#f39d3c',
  'pf-blue': '#0088ce',
  'pf-black': '#030303'
};
```
![legend](https://user-images.githubusercontent.com/3019213/40485286-66097e88-5f5e-11e8-9220-632941068746.gif)

serenamarie125 commented:
@aljesusg, pf-orange-400 (#ec7a08) pf-red-100 (#cc0000) pf-green-400 (#3f9c35).

serenamarie125 commented:
I think that the legend should show up on the bottom left by default.
The items in the legend should have arrows, just show the line, and please make the line thickness thinner.

aljesusg commented:
Ok, I saw that in the graph they are using the base colors. Should I change the colors of graph too?

serenamarie125 commented:
Also, that mock is old :) Here's the latest.  We changed the color from blue to green 

<img width="364" alt="screen shot 2018-05-24 at 8 33 07 am" src="https://user-images.githubusercontent.com/12504403/40485656-242fae8c-5f2d-11e8-9e6b-b94dfd19365b.png">

abonas commented:
> The items in the legend should have arrows, just show the line, and please make the line thickness thinner.

@serenamarie125 
'should have arrows' means should **not** have arrows, right? :)

aljesusg commented:
Thanks @serenamarie125 
I changed the app colors to PF colors, about arrows? ^^

serenamarie125 commented:
sorry bout that @aljesusg !  Missed a word
The items in the legend should *not* have arrows, just show the line, and please make the line thickness thinner.

aljesusg commented:
![legend](https://user-images.githubusercontent.com/3019213/40486531-dbd01aca-5f61-11e8-9bd3-eca34ecf0e47.gif)

Last release @abonas & @serenamarie125

abonas commented:
@aljesusg looks great!
please note @serenamarie125 comment about legend's default position: "the legend should show up on the bottom left by default."

aljesusg commented:
![legend](https://user-images.githubusercontent.com/3019213/40498227-baaea556-5f7e-11e8-8fd4-3608c977223e.gif)

Last release, @abonas @serenamarie125 please review UI

mtho11 commented on src/pages/ServiceGraph/ServiceGraphPage.tsx:
```diff
@@ -37,7 +40,8 @@ export default class ServiceGraphPage extends React.Component<ServiceGraphPagePr
     this.state = {
-      summaryData: { summaryType: 'graph', summaryTarget: null }
+      summaryData: { summaryType: 'graph', summaryTarget: null },
+      hideLegend: true
```
Please put `hideLegend` into Redux store. The `state` for _ServiceGraphPage_ has actually already been removed and _ServiceGraphPage_ is a stateless component once the PR https://github.com/kiali/kiali-ui/pull/354 is merged.

mtho11 commented on src/pages/ServiceGraph/ServiceGraphPage.tsx:
```diff
@@ -78,6 +82,10 @@ export default class ServiceGraphPage extends React.Component<ServiceGraphPagePr
   };
 
+  handleLegendClick = () => {
+    this.setState({ hideLegend: !this.state.hideLegend });
```
Please dispatch a Redux Action instead

aljesusg commented on src/components/GraphFilter/GraphLegend.tsx:
```diff
@@ -0,0 +1,92 @@
+
+  render() {
+    return (
+      <div style={{ marginTop: 20, clear: 'both' }}>
```
wowow i didn't know that you added Typestyle. perfect thanks

aljesusg commented:
@mtho11 I did the changes

aljesusg commented on src/components/GraphFilter/GraphLegend.tsx:
```diff
@@ -0,0 +1,92 @@
+  dashed: boolean;
+}
+
+class Arrow extends React.Component<ArrowProps> {
```
Ok, I'll do this changes while I wait @serenamarie125 review thanks @mtho11

aljesusg commented on src/components/GraphFilter/GraphLegend.tsx:
```diff
@@ -0,0 +1,92 @@
+        height: '5px',
+        float: 'left',
+        backgroundImage:
+          'repeating-linear-gradient(90deg, ' +
```
Yes more cleaner now ^^

aljesusg commented:
Thanks @mtho11
